tests/test_analysis.py

```diff
@@ -623,46 +647,214 @@ def helper_test_update_harmonic_vibe_props(analysis):
             "Harmonic vibe damping value is invalid for harmonic vibe properties 0: foo"
         )
 
-    if analysis._is_connection_up():
-        try:
-            analysis.update_harmonic_vibe_props(
-                "Tutorial Project",
-                [
-                    {
-                        "cca_name": "Main Board",
-                        "harmonic_vibe_count": 2,
-                        "harmonic_vibe_damping": "0.01, 0.02",
-                        "part_validation_enabled": False,
-                        "require_material_assignment_enabled": False,
-                        "analysis_temp": 20,
-                        "analysis_temp_units": "foo",
-                        "filter_by_event_frequency": False,
-                    },
-                ],
-            )
-            pytest.fail("No exception raised when using an invalid parameter")
-        except Exception as e:
-            assert type(e) == SherlockUpdateHarmonicVibePropsError
+    try:
+        result = analysis.update_harmonic_vibe_props(
+            "Tutorial Project",
+            [
+                {
+                    "cca_name": "Main Board",
+                    "harmonic_vibe_count": 2,
+                    "harmonic_vibe_damping": "0.01, 0.02",
+                    "part_validation_enabled": False,
+                    "require_material_assignment_enabled": False,
+                    "analysis_temp": 20,
+                    "analysis_temp_units": "C",
+                    "filter_by_event_frequency": False,
+                },
+            ],
+        )
```

Comment:
This test should only run when connected.
Validate the result value indicates that the update was successful here instead of after checking if connected.
Move this to below the test that validates the exception type since that is a prerequisite for catching this exception.